unblob/processing.py

```diff
@@ -119,3 +149,49 @@ def calculate_unknown_chunks(
         unknown_chunks.append(unknown_chunk)
 
     return unknown_chunks
+
+
+def calculate_entropy(path: Path, *, draw_plot: bool):
+    """Calculate and log shannon entropy divided by 8 for the file in 1mB chunks."""
+    logger.info("Calculating entropy for file", path=path)
+    percentages = []
+
+    # We could use the chunk size instead of another syscall,
+    # but we rely on the actual file size written to the disk
+    file_size = path.stat().st_size
+
+    # Smaller chuk size would be very slow to calculate. This takes ~ 3sec for a 4,5 GB file.
```

Comment:
What if we cut the file into a fixed amount of chunks, with lower and upper limit to the chunk size (say the upper limit at 1MB and the lower limit at 1KB) ? This would provide more granularity to users looking into small files (e.g. encrypted config file) but keep a "constant" time regardless of the file being analyzed.

Reply:
Seems like a very good idea

Reply:
I did this:

```python
def calculate_buffer_size(
    file_size, *, chunk_count: int, min_limit: int, max_limit: int
) -> int:
    """Split the file into even sized chunks, limited by lower and upper values."""
    # We don't care about floating point precision here
    buffer_size = file_size // chunk_count
    buffer_size = max(min_limit, buffer_size)
    buffer_size = min(buffer_size, max_limit)
    return buffer_size
```